servicetalk-traffic-resilience-http/src/main/java/io/servicetalk/traffic/resilience/http/AbstractTrafficResilienceHttpFilter.java

```diff
@@ -249,37 +282,101 @@ private Single<StreamingHttpResponse> handleAllow(
                     }
                     return Single.succeeded(resp).shareContextOnSubscribe();
                 })
-                .liftSync(new BeforeFinallyHttpOperator(new TerminalSignalConsumer() {
-                    @Override
-                    public void onComplete() {
-                        try {
-                            if (breaker != null) {
-                                breaker.onSuccess(nanoTime() - startTimeNs, NANOSECONDS);
-                            }
-                        } finally {
-                            onSuccessTicketTerminal.accept(ticket);
-                            ticketObserver.onComplete();
-                        }
-                    }
+                .liftSync(new BeforeFinallyHttpOperator(
+                        new SignalConsumer(this, ticket, ticketObserver, breaker, startTimeNs)));
+    }
 
-                    @Override
-                    public void onError(final Throwable throwable) {
-                        AbstractTrafficResilienceHttpFilter.this.onError(throwable, breaker, startTimeNs, ticket);
-                        ticketObserver.onError(throwable);
+    private Single<StreamingHttpResponse> dryRunHandleAllow(
+            final Function<StreamingHttpRequest, Single<StreamingHttpResponse>> delegate,
+            final Function<HttpResponseMetaData, Duration> delayProvider, final StreamingHttpRequest request,
+            final Ticket ticket,
+            final TicketObserver ticketObserver, @Nullable final CircuitBreaker breaker, final long startTimeNs) {
+        SignalConsumer signalConsumer = new SignalConsumer(
+                this, ticket, ticketObserver, breaker, startTimeNs);
+        return delegate.apply(request)
+                // This logic has the same general structure as the `handleAllow` case, but there is a twist: we want
+                // to always return the successful single even when we fail the predicates. Because we use a latch in
+                // SignalConsumer it's safe to signal those conditions eagerly in the `.flatMap` operation and let the
+                // request proceed as usual and any following signals (such as from body processing) will be ignored.
+                .flatMap(resp -> {
```

Comment:
Bcz there is nothing async inside this variant (like draining), consider using `map` instead of `flatMap`